### ISSUE TITLE
Only show standalone debug messages in debug mode

### ISSUE THREAD

**Issue by mstimberg** [brian-team/brian2#882]
Currently, standalone mode shows the "Number of synapses: ..." and "Number of spikes: ..." messages by default. While the functions printing this information are called `_debugmsg_`, they are printed independently of the `debug` flag (which is off by default). Currently, the only method to switch them off is to use `with_output=False` which switches off *all* output, including the progress reporting (as a side note: wouldn't it make sense to merge `with_output` and `debug` into one option?). You can work around this by using `with_output=False` and `report='stderr'`, but this is a bit more complicated than it should be.

To sum up: I propose to show the standalone debug messages only when `debug=True` has been set.
